src/components/data-management/SpeciesSelector.jsx

```diff
@@ -3,20 +3,38 @@ import PropTypes from 'prop-types';
 import {
   Select, Typography, Space, Divider, Skeleton,
 } from 'antd';
-import _ from 'lodash';
 
 const { Text } = Typography;
 
 const SpeciesSelector = (props) => {
-  const { data, onChange } = props;
+  const { data, value, onChange } = props;
 
-  if (!data) {
+  if (!data || data.length === 0) {
     return <Skeleton.Input style={{ width: 300 }} size='small' />;
   }
 
+  const createLabel = (scientificName, displayName) => (
```

Comment:
This is a bit messy and unnecessarily coupled.  `createLabel` returns a React component, `labeledValue` is coupled with `createLabel` and calls it if there is no `organismId`, `organismId` is actually equal to that `value` that you have in default props. I would suggest:

1. Separation of concerns: Either make `labeledValue` to be only about returning organismId and rename it to `getOrganismId`. Or combine functions `createLabel` and `labeledValue` into 1 function if these 2 things have to be called together each time and rename that function to show better what it is actually doing.
2. if `value` in props is actually organismId, then why don't you rename it to `organismId`? `value` is super ambigious. Also, if it is in props,  why do you need to pass it in the `labeledValue` function?

Reply:
Cleaned up by removing `labelInValue`